expected-consensus.md

```diff
@@ -144,132 +145,158 @@ the prior block, or "losing" tickets generated in this epoch.
 
-Succinctly, the process of crafting a new `ticket` and `ElectionProof` in round `N` is as follows:
+Succinctly, the process of crafting a new `Ticket` in round `N` is as follows. We use:
+
+- The ECVRF algorithm from [Goldberg et al. Section 5](https://tools.ietf.org/html/draft-irtf-cfrg-vrf-04#page-10), with:
+  - Sha256 for our hashing function
+  - Secp256k1 for our curve
+  - Note that the operation type in step 2.1 is necessary to prevent an adversary from guessing an election proof for a miner ahead of time.
+- The TODO VDF impl.
 
 ```text
-Input: {miners}, VDF, parentTickets at round N-K, Power Table, a sorting algorithm Sort, a signature algorithm Sig, a hash function H
-Output: T, W
-
-1. Generate a ticket `T`
-  i. L <-- Sort(parentTickets)
-  ii. l <-- min(L)
-  iii. pi, v <-- VDF(H(l))
-  iv. T <-- Sig(pi, v)
-2. Generate an ElectionProof = W
-  i. T_o <-- DrawMinTicket(L round back)
-  ii. W <-- VRF(SK, T_o)
-3. Return: T, W
+Input: parentTickets at round N-1, miner's private key SK
+Output: newTicket
+
+0. Prepare new ticket
+	i. newTicket <-- New()
+1. Draw prior ticket
+	i. 	 # take the last ticket for each parent 'Tickets' array
+			lastTickets <-- map(parentTickets, fun(x): x.last)
+	ii.  # sort these tickets 
+  			sortedTickets <-- Sort(lastTickets)
+    iii. # draw the smallest ticket
+  			parentTicket <-- min(sortedTickets)
+2. Run it through VRF and get determinstic output
+	i.   # take the VDFOutput of that ticket as input, specifying the personalization (see data-structures)
+			input <-- VRFPersonalization.Ticket | parentTicket.VDFOutput
+	ii.	 # run it through the VRF and store the VRFProof in the new ticket
+			newTicket.VRFProof <-- ECVRF_prove(SK, input)
+	iii. # draw a deterministic output from this
+			VRFOutput <-- ECVRF_proof_to_hash(newTicket.VRFProof)
+3. Run that deterministic output through a VDF
+    i.  # run eval with our VDF and its evaluation k on VRFOutput
+  			y, pi <-- Eval(ek, VRFOutput)
+    ii. # Store the output and proof in our ticket
+  			newTicket.VDFOutput <-- y
+  			newTicket.VDFProof 	<-- pi
+4. Return the new ticket
 ```
 
-#### Checking election results
+### Checking election results
+
+Now, a miner must also check whether they are eligible to mine a block in this round. For how Election Proofs are validated, see [election validation](mining.md#election-validation).
 
-Now, a miner must also check whether they have won in this round.
+To do so, the miner will use tickets from K blocks back as randomness to uniformly draw a value from 0 to 1. Comparing this value to their power, they determine whether they are eligible to mine. A user's `power` is defined as the ratio of the amount of storage they proved as of their last PoSt submission to the total storage in the network as of the current block.
```

Comment:
With the appropriate bookkeeping in the miner actor, the storage proven at last PoSt should be the same value as their storage power in the current block, so both can be sampled from the same point. The devil is in "appropriate bookkeeping".